collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceActivity.java

```diff
@@ -191,13 +214,47 @@ public void initMap(MapFragment newMapFragment) {
         }
         helper.setBasemap();
 
+        locationStatus = findViewById(R.id.geotrace_location_status);
+        collectionStatus = findViewById(R.id.geotrace_collection_status);
         traceSettingsView = getLayoutInflater().inflate(R.layout.geotrace_dialog, null);
-        RadioGroup group = traceSettingsView.findViewById(R.id.radio_group);
-        group.check(group.getChildAt(recordingMode).getId());
-        timeDelay = traceSettingsView.findViewById(R.id.trace_delay);
-        timeDelay.setSelection(restoredTimeDelayIndex);
-        timeUnits = traceSettingsView.findViewById(R.id.trace_scale);
-        timeUnits.setSelection(restoredTimeUnitsIndex);
+        radioGroup = traceSettingsView.findViewById(R.id.radio_group);
+        radioGroup.setOnCheckedChangeListener(this::updateRecordingMode);
+        autoOptions = traceSettingsView.findViewById(R.id.auto_options);
+        autoInterval = traceSettingsView.findViewById(R.id.auto_interval);
+        autoInterval.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                intervalIndex = position;
+            }
+
+            @Override public void onNothingSelected(AdapterView<?> parent) { }
+        });
+
+        // Populate the auto_interval spinner with the options in INTERVAL_OPTIONS.
+        String[] options = new String[INTERVAL_OPTIONS.length];
+        for (int i = 0; i < INTERVAL_OPTIONS.length; i++) {
+            options[i] = formatInterval(INTERVAL_OPTIONS[i]);
+        }
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, options);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        autoInterval.setAdapter(adapter);
+
+        accuracyThreshold = traceSettingsView.findViewById(R.id.accuracy_threshold);
+        accuracyThreshold.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                accuracyThresholdIndex = position;
+            }
+
+            @Override public void onNothingSelected(AdapterView<?> parent) { }
+        });
+
+        // Populate the accuracy_threshold spinner with the options in ACCURACY_THRESHOLD_OPTIONS.
+        options = new String[ACCURACY_THRESHOLD_OPTIONS.length];
+        for (int i = 0; i < ACCURACY_THRESHOLD_OPTIONS.length; i++) {
+            options[i] = formatAccuracyThreshold(ACCURACY_THRESHOLD_OPTIONS[i]);
+        }
+        adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, options);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        accuracyThreshold.setAdapter(adapter);
```

Comment:
This is mostly duplicated logic that would be nice to put into a utility method that takes e.g. `String[] options` and returns `adapter`.

Reply:
Yeah, that's better.  Done.